### PR TITLE
Adds water overlay to water tiles

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Mobs will now appear to be halfway submerged when entering water tiles
/:cl:

water

Also adds slowdown to deep water

### CODE REVIEW

**Comment by KorPhaeron** [comment]
If I was smart I'd subtract colour/transparency from the bottom half of the underlying sprite so it balances out with the overlay over the bottom half and hide that ugly line

Is anyone here smart enough to do that in the next 2 hours?

**Comment by KorPhaeron** [comment]
The overlay is casting a weird shadow though which is making that line

**Comment by MMMiracles** [comment]
Wouldn't this cause a weird shutter effect when moving up and down, since it's just an effect on the turf? Wouldn't it be better if it just applied an overlay on the mob and got removed when they left the water turf?

**Comment by KorPhaeron** [comment]
I guess I could just make the overlay full tile

**Comment by KorPhaeron** [comment]
@deathride58 is there an easy way around the occlusion shadow

**Comment by nicbn** [comment]
Just remove ambient occlusion

**Comment by deathride58** [comment]
@KorPhaeron You could try moving the overlay up one single plane. Either that, or apply the overlay to the mob itself. Since Byond doesn't expose HLSL shader controls to devs, there's no way to have fine control over what does/does not get affected by the AO filter

File: code/game/turfs/simulated/floor/plating/misc_plating.dm [hunk 1]
```diff
@@ -118,6 +118,23 @@
 	desc = "You get the feeling that nobody's bothered to actually make this water functional..."
 	icon_state = "water"
 	baseturfs = /turf/open/floor/plating/beach/water
+	slowdown = 2
+	var/obj/effect/overlay/water/water
+
+/turf/open/floor/plating/beach/water/Initialize()
```
Comment (ShizCalev) [inline code/game/turfs/simulated/floor/plating/misc_plating.dm]: could you add this to` /turf/open/floor/holofloor/beach/water` & `/turf/open/water` too?

**Comment by MrDoomBringer** [comment]
You should have a mob specific overlay, as well as one for the rest of the things (To prevent the shutter effect), as players will take up the vast majority of moving objects in water. Its not great code-wise, but it would look a lot better in practice imo

**Comment by MrStonedOne** [comment]
What fucking shutter effect?

**Comment by SpaceManiac** [comment]
If the overlay takes up only the bottom half of the tile, moving between tiles vertically will look goofy.
image

**Comment by optimumtact** [comment]
the answer is the smoothing system but with teh water overlays

think about it

**Comment by SpaceManiac** [comment]
smoothing would "solve" that specific problem but it wouldn't make the overlay look any better vs. just having it be full-tile

**Comment by KorPhaeron** [comment]
Putting it on the mob would look weird with mobs larger than 32x32 and we'd have to do it for every atom that entered so guns wouldn't just float etc

File: code/game/turfs/simulated/floor/plating/misc_plating.dm (hunk 1 above)
Comment (KorPhaeron) [reply code/game/turfs/simulated/floor/plating/misc_plating.dm]: For the holotiles sure, dont think this would look good with our other water tiles

**Comment by nicbn** [comment]
Do it for every movable atom then

**Comment by optimumtact** [comment]
honestly the correct answer is probably to do a full tile overlay and have a transition tile

**Comment by optimumtact** [comment]
whoops

File: code/game/turfs/simulated/floor/plating/misc_plating.dm (hunk 1 above)
Comment (ShizCalev) [reply code/game/turfs/simulated/floor/plating/misc_plating.dm]: holotiles plz

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by MrStonedOne** [comment]
wave filter when

**Comment by SpaceManiac** [comment]
I spent some time fiddling with this and couldn't get it to look good.

The only thing even remotely passable is full-tile overlay, with `plane = BLACKNESS_PLANE` to stave off the drop shadow.

**Comment by optimumtact** [comment]
I think the only way it works is if the overlay is on the mob and then animates down to the ground and stops at the water edge when they enter a transition from water -> normal land (as if they're leaving water)

**Comment by JJRcop** [comment]
It would look fine if the overlay went behind mobs as their bottom pixel leaves it.

**Comment by SpaceManiac** [comment]
Anyone who actually knows how to *implement* any of these suggestions is welcome to try it, in the meantime this PR is getting nowhere. Reopen at your leisure.

**Pull request closed without merging**